crates/rolldown/src/hmr/hmr_ast_finalizer.rs

```diff
@@ -55,6 +57,9 @@ pub struct HmrAstFinalizer<'me, 'ast> {
   pub dependencies: FxIndexSet<ModuleIdx>,
   pub imports: FxHashSet<ModuleIdx>,
   pub generated_static_import_infos: FxHashMap<ModuleIdx, String>,
+
+  // We need to store the static import statements for external separately, so we could put them outside of the `try` block.
+  pub generated_static_import_stmts_from_external: FxIndexMap<ModuleIdx, ast::Statement<'ast>>,
```

Comment:
```suggestion
  generated_static_import_stmts_from_external: FxIndexMap<ModuleIdx, ast::Statement<'ast>>,
```
Maybe it's ok to keep it private? It would reduce the cognitive load as we don't need to think about the possibility that it is modified from outside.

Reply:
Good suggestion!